### ISSUE TITLE
Deploy only for merged PRs, not every closed PR

### ISSUE THREAD

**Issue by Aprillion** [StampyAI/stampy-ui#129]
https://github.com/StampyAI/stampy-ui/actions/runs/3917865363 was incorrectly run after closing PR by declining it.

Let's update GH actions to only run the Deploy for *merged* PRs and only to the master branch.

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges

**Comment by Aprillion** [comment]
Aaah, fixing GH actions was a bumpy ride for me, but looks like https://github.com/StampyAI/stampy-ui/actions/runs/3974800375 is finally green.

**Comment by ccstan99** [comment]
Hmm... doesn't seem to be working. It just got triggered when I created my PR, supposedly when I pushed a commit?! https://github.com/StampyAI/stampy-ui/actions/runs/3977645920/workflow. It's weird because I'm wasn't working on the master branch or closing a pull request.

The syntax looks funny to me. `if_merged` is the name of the step. I think the line checking ` if: github.event.pull_request.merged == true` is the important line we need.
```
jobs:
  if_merged:
    if: github.event.pull_request.merged == true
    runs-on: ubuntu-latest
```
